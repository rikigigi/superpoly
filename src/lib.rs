pub mod poly;

#[cfg(test)]
mod tests {
    use crate::poly::*;
    #[test]
    fn test_poly1(){
       let p = Poly::new(vec![3.0,4.55,4.3,5.02,7.1],vec![0.0,0.0,0.0,9.0,90.0], 1.0, 3.0);
       insta::assert_yaml_snapshot!(p.a.iter().map(|x| x).collect::<Vec<&f64> >() );
       insta::assert_yaml_snapshot!( (-350..351).map(|x| p.calc(x as f64 /100.0 ).0 ).collect::<Vec<f64> >());
       insta::assert_yaml_snapshot!( (-350..351).map(|x| p.calc(x as f64 /100.0 ).1 ).collect::<Vec<f64> >());
       insta::assert_yaml_snapshot!( (-350..351).map(|x| p.calc(x as f64 /100.0 ).2 ).collect::<Vec<f64> >());


    }
}
