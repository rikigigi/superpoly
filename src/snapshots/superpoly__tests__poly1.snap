---
source: src/lib.rs
expression: "p.a.iter().map(|x| x).collect::<Vec<&f64>>()"
---
- 0.7078268432974726
- -0.24875927787167154
- 0.03971775421403232
- -0.0035106650653714266
- 0.00018008935475770878
- -0.0000052883292269523435
- 0.00000008137166246762725
- -0.0000000004971001717684232

